src/components/TokenEnterAmount.tsx

```diff
@@ -35,6 +41,200 @@ function groupNumber(value: string) {
   return value.replace(/\B(?=(\d{3})+(?!\d))(?<!\.\d*)/g, 'group')
 }
 
+function roundTokenAmount(value: string, token: TokenBalance) {
+  const { decimalSeparator, groupingSeparator } = getNumberFormatSettings()
+  if (value === '') {
+    return ''
+  }
+
+  const bigNum = parseInputAmount(value, decimalSeparator)
+
+  if (bigNum.isLessThan(0.000001)) {
+    return `<0${decimalSeparator}000001 ${token.symbol}`
+  }
+
+  const grouped = groupNumber(bigNum.decimalPlaces(6).toString())
+    .replaceAll('.', decimalSeparator)
+    .replaceAll('group', groupingSeparator)
+  return `${grouped} ${token.symbol}`
+}
+
+function roundLocalAmount(value: string, localCurrencySymbol: LocalCurrencySymbol) {
+  const { decimalSeparator, groupingSeparator } = getNumberFormatSettings()
+  if (value === '') {
+    return ''
+  }
+
+  const bigNum = parseInputAmount(value, decimalSeparator)
+
+  if (bigNum.isLessThan(0.000001)) {
+    return `<${localCurrencySymbol}0${decimalSeparator}000001`
+  }
+
+  const rounded = bigNum.isLessThan(0.01) ? bigNum.toPrecision(1) : bigNum.toFixed(2)
+  const grouped = groupNumber(rounded.toString())
+    .replaceAll('.', decimalSeparator)
+    .replaceAll('group', groupingSeparator)
+  return `${localCurrencySymbol}${grouped}`
+}
+
+export function useEnterAmount(props: {
+  token: TokenBalance
+  onSelectToken?: (token: TokenBalance) => void
+}) {
+  const { decimalSeparator, groupingSeparator } = getNumberFormatSettings()
+  const inputRef = useRef<RNTextInput>(null)
+  const bottomSheetRef = useRef<BottomSheetModalRefType>(null)
+  const [amount, setAmount] = useState('')
+  const [amountType, setAmountType] = useState<AmountEnteredIn>('token')
+
+  // this should never be null, just adding a default to make TS happy
+  const localCurrencySymbol = useSelector(getLocalCurrencySymbol) ?? LocalCurrencySymbol.USD
+  const tokenInfo = useTokenInfo(props.token.tokenId)
+  const usdToLocalRate = useSelector(usdToLocalCurrencyRateSelector)
+
+  const amountRaw = useMemo(() => {
+    return amount.replaceAll(groupingSeparator, '').replaceAll(decimalSeparator, '.')
+  }, [amount])
+
+  const derived = useMemo(() => {
+    if (amountType === 'token') {
+      const parsedTokenAmount = parseInputAmount(amount, decimalSeparator)
+      const tokenToLocal = convertTokenToLocalAmount({
+        tokenAmount: parsedTokenAmount,
+        tokenInfo,
+        usdToLocalRate,
+      })
+      const convertedTokenToLocal =
+        tokenToLocal && tokenToLocal.gt(0)
+          ? tokenToLocal.toString().replaceAll('.', decimalSeparator)
+          : ''
+      const parsedLocalAmount = parseInputAmount(
+        convertedTokenToLocal.replaceAll(groupingSeparator, ''),
+        decimalSeparator
+      )
+
+      return {
+        token: {
+          amount: amount,
+          bignum: parsedTokenAmount,
+          readable: roundTokenAmount(amount, props.token),
```

Comment:
- amount is for the actual value
- bignum is for all the processing (previously it was called `localToToken` or `parsedTokenAmount` depending on what type of input it was referring to so it is move unified now while keeping a more understandable name)
- readable is specifically for `TokenEnterAmount` as that component is supposed to show only two things in regards to any amounts:
  - actual input value
  - converted amount